SoftLayer/managers/hardware.py

```diff
@@ -257,6 +257,28 @@ def rescue(self, hardware_id):
         """
         return self.hardware.bootToRescueLayer(id=hardware_id)
 
+    def trunk(self, hardware_id, public, vlan_id):
+        """Add vlan trunk to a server.
+
+        :param integer hardware_id: the hardware ID to add vlan trunk
+        :param integer vlan_id: the vlan number
+
+        Example::
+
+            result = mgr.trunk(824671, 1234)
+        """
+        hw = self.get_hardware(hardware_id)
+        primary = 'primaryBackendIpAddress'
+        if public:
+            primary = 'primaryFrontendIpAddress'
+        cids = [c['id'] for c in hw['networkComponents']
+                if c.get('primaryIpAddress') == hw[primary]]
+        param = [{'vlanNumber': int(vlan_id)}]
```

Comment:
The documentation expects an integer for `vlan_id`. Casting like this should not be needed.